### DISCUSSION TITLE
Tracking utm_ parameters with redirects

### DISCUSSION THREAD

**Discussion by gravelld** [plausible/analytics#2630] [Anything Else]
I'm failing to track utm_ parameters and I wondered if there were any further obvious gotchas I'm missing...

I send a URL out to a mailing list with this query string:

    ?utm_medium=email&utm_source=blah&utm_campaign=lifecycle-not-added-storage

If the link is clicked, and when the user hits our server, they are 302 redirected to log into the server. The UTM parameters are included in the 301 redirect, so the eventual page that is shown has a query string similar to:

    ?next=/original-requested-page&utm_medium=email&utm_source=blah&utm_campaign=lifecycle-not-added-storage

However, since sending these emails out six weeks ago I have not had a *single* hit for a utm_ parameter from anyone else! I've sent about 500 emails and received replies to the emails - I'm pretty sure they are being delivered.

If I run the test myself and I copy/paste the second URL above I _think_ I have seen this work. But that's a slightly different test because there's no 302 redirect taking place - it's just a direct GET for that URL which is correctly resolved.

Are there any gotchas around redirects I need to be mindful of?

**Comment by metmarkosaric** [comment]
hi @gravelld! it's difficult to answer without having the exact urls but the issue could be that you have the `?next` in the url. by default, we strip all the query parameters except for utms. you can [read more here](https://plausible.io/docs/custom-query-params). you can also contact [our customer support](https://plausible.io/contact) with actual urls and we'll check

**Comment by gravelld** [comment]
Thank you so much - I have emailed and if there's anything of general use I shall endeavor to record it here.
